packages/autorest.python/autorest/codegen/serializers/sample_serializer.py

```diff
@@ -0,0 +1,144 @@
+# pylint: disable=too-many-lines
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import logging
+from typing import Dict, Any
+from jinja2 import Environment
+
+from autorest.codegen.models.credential_types import AzureKeyCredentialType
+from autorest.codegen.models.credential_types import TokenCredentialType
+from autorest.codegen.models.imports import FileImport, ImportType
+from autorest.codegen.models.operation import OperationBase
+from autorest.codegen.models.operation_group import OperationGroup
+from autorest.codegen.serializers.import_serializer import FileImportSerializer
+from ..models import CodeModel
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SampleSerializer:
+    def __init__(
+        self,
+        code_model: CodeModel,
+        env: Environment,
+        operation_group: OperationGroup,
+        operation: OperationBase[Any],
+        sample: Dict[str, Any],
+        file_name: str,
+        sample_origin_name: str,
+    ) -> None:
+        self.code_model = code_model
+        self.env = env
+        self.operation_group = operation_group
+        self.operation = operation
+        self.sample = sample
+        self.file_name = file_name
+        self.sample_origin_name = sample_origin_name
+
+    def _imports(self) -> FileImportSerializer:
+        imports = FileImport()
+        namespace = (self.code_model.options["package_name"] or "").replace(
+            "-", "."
+        ) or self.code_model.namespace
+        imports.add_submodule_import(
+            namespace, self.code_model.client.name, ImportType.THIRDPARTY
+        )
+        credential_type = getattr(self.code_model.credential, "type", None)
```

Comment:
I think we can directly grab the imports needed off of `code_model.credential`

Reply:
@iscai-msft we need to distinguish the type of credential. If "tokencredential", import from "azure.identity";if "keycredential", import from "azure.core"; if None, no import

Reply:
yeah that `credential` property will return the credential object, and we can get the import off of `credential.imports()`

Reply:
@iscai-msft  `credential.imports()` only work for keycredential. For tokencredential, we need like `from azure.identity import DefaultAzureCredential` but `.imports()` returns `from azure.core.credentials imports TokenCredential`. So we still need the logic for import